### ISSUE TITLE
Installing with pipx could be made easier

### ISSUE THREAD

**Issue by Steve-Tech** [pwr-Solaar/Solaar#2147]
**Information**

- Solaar version: solaar 1.1.10
- Distribution: Debian Trixie/Testing
- Kernel version: `Linux 6.5.0-1-amd64 x86_64 GNU/Linux`
- Output of `solaar show`: `solaar: command not found`

- Contents of `~/.config/solaar/config.yaml` (or `~/.config/solaar/config.json` if `~/.config/solaar/config.yaml` not present): N/A

- Errors or warrnings from Solaar: See description

**Describe the bug**
Most distros warn against using pip now, so something like pipx is needed. But there's a bunch of dependency issues, solving them would make it much easier to install (but I can settle for documentation):

**First error**: `ERROR: Dependency "dbus-1" not found, tried pkgconfig`
<details>
<summary>Full Error</summary>

```
$ pipx install solaar
Fatal error from pip prevented installation. Full pip output in file:
/home/stephen/.local/pipx/logs/cmd_2023-09-27_17.26.14_pip_errors.log

pip seemed to fail to build package:
dbus-python>=1.3.2

Some possibly relevant errors from pip install:
error: subprocess-exited-with-error
../subprojects/dbus-gmain/meson.build:107:11: ERROR: Dependency lookup for dbus-1 with method 'pkgconfig' failed: Pkg-config binary for machine 1 not found. Giving up.
error: metadata-generation-failed

Error installing solaar.
```
</details>

Solved with `sudo apt install libdbus-1-dev`

**Second error**: `ERROR: Dependency "glib-2.0" not found, tried pkgconfig`
<details>
<summary>Full Error</summary>

```
$ pipx install solaar
Fatal error from pip prevented installation. Full pip output in file:
/home/stephen/.local/pipx/logs/cmd_2023-09-27_17.30.07_pip_errors.log

pip seemed to fail to build package:
dbus-python>=1.3.2

Some possibly relevant errors from pip install:
error: subprocess-exited-with-error
../subprojects/dbus-gmain/meson.build:108:11: ERROR: Dependency "glib-2.0" not found, tried pkgconfig
error: metadata-generation-failed

Error installing solaar.
```
</details>

Solved with `sudo apt install libglib2.0-dev`

**Third error**: `missing required system package python3-gi (in Ubuntu) or python3-gobject (in Fedora)`
<details>
<summary>Full Error</summary>

```
$ pipx install solaar
installed package solaar 1.1.10, installed using Python 3.11.5
These apps are now globally available
- solaar
done! ✨ 🌟 ✨
$ solaar -d
solaar: missing required system package python3-gi (in Ubuntu) or python3-gobject (in Fedora)
```
</details>

`python3-gi` was already installed, but the following worked:
```sh
sudo apt install libcairo2-dev libgirepository1.0-dev
pipx inject solaar PyGObject
```

**To Reproduce**
1. Install pipx
2. Run `pipx install solaar`

**Screenshots**
N/A

**Additional context**
TL;DR: If you're trying to install solaar using pipx on debian, run this:
```sh
sudo apt install libdbus-1-dev libglib2.0-dev libcairo2-dev libgirepository1.0-dev
pipx install solaar[report-descriptor]
pipx inject solaar PyGObject
```

**Comment by pfps** [comment]
Solaar is not set up to be installed using pipx.

**Comment by pfps** [comment]
It turns out that pipx can be easily used to install Solaar, so long as you use `--system-site-packages`.  I think that this is the right way to go and will update the Solaar documentation.

**Comment by Steve-Tech** [comment]
Hi @pfps, I really appreciate you having a look at this issue again, but I just tested `--system-site-packages` in an Ubuntu 22.04 VM I had, and it still has the first 2 errors, (solved with `sudo apt install libdbus-1-dev libglib2.0-dev`). Although interestingly, rolling back and installing with bare pip seems to also has similar issues, with the addition of missing `meson`, `ninja`, and `patchelf` from the `dbus-python` side? So I also guess this isn't really a pipx specific issue anymore, and probably more of a `dbus-python` issue too.

**Comment by pfps** [comment]
Solaar needs some system packages installed.   As far as I know these are not Python packages so it doesn't look to me as if pip or pipx can install them.

**Comment by Steve-Tech** [comment]
Yeah that's fair enough, thanks for your help!
